### PR TITLE
[Feat] Wert

### PR DESCRIPTION
**Pull request by JulesGuesnon**
# Description
Added Wert to the fund page

### CODE REVIEW

File: pages/[walletAddress]/fund-wallet.tsx [hunk 1]
```diff
@@ -38,15 +94,38 @@ const TopUpPage = () => {
       </div>
       <main className="min-h-fit grow">
         <div className="mx-auto min-h-full max-w-7xl px-4 py-6 sm:px-6 lg:px-8">
+          {PREFERED_NETWORK !== NetworkType.MAINNET && (
+            <div>
+              <h2 className="text-xl text-white">Buy crypto</h2>
+              <p className="mt-2 text-zinc-200">
+                Our provider {"doesn't"} support transferring to Tezos contract
+                yet. So after the transaction succeed, we will automatically
```
Comment (rueshyna) [inline pages/[walletAddress]/fund-wallet.tsx:102]:
```suggestion
                yet. Therefore, after the transaction succeeds, we will automatically
```

File: pages/[walletAddress]/fund-wallet.tsx [hunk 2]
```diff
@@ -38,15 +94,38 @@ const TopUpPage = () => {
...
             <h2 className="text-center text-xl text-zinc-600">
               Please select a wallet in the sidebar
             </h2>
           ) : (
-            <TopUp
-              address={state.currentContract ?? ""}
-              closeModal={() => {}}
-            />
+            <>
+              <h2 className="mt-12 text-xl text-white">Send to the contract</h2>
```
Comment (rueshyna) [inline pages/[walletAddress]/fund-wallet.tsx:123]:
```suggestion
              <h2 className="mt-12 text-xl text-white">Fund from {alias(logged_in_account)}</h2>
```
something like this

File: pages/[walletAddress]/fund-wallet.tsx [hunk 3]
```diff
@@ -1,13 +1,69 @@
...
+
+  const onSuccess = async (txId: string) => {
+    if (!state.currentContract) return;
+
+    try {
+      const transaction = await fetch(
+        `${TZKT_API_URL}/v1/operations/${txId}`
+      ).then(res => res.json());
+
+      if (!transaction || transaction.length === 0) {
```
Comment (rueshyna) [inline pages/[walletAddress]/fund-wallet.tsx:27]: need to check the block's level. at least, need to already pass one block/confirmation.

File: pages/[walletAddress]/fund-wallet.tsx [hunk 4]
```diff
@@ -38,15 +94,38 @@ const TopUpPage = () => {
       </div>
       <main className="min-h-fit grow">
         <div className="mx-auto min-h-full max-w-7xl px-4 py-6 sm:px-6 lg:px-8">
+          {PREFERED_NETWORK !== NetworkType.MAINNET && (
+            <div>
+              <h2 className="text-xl text-white">Buy crypto</h2>
```
[the remark this answers is not in the document]
Comment (rueshyna) [reply pages/[walletAddress]/fund-wallet.tsx:99]: - the style of two `h2` is different.
 - I think "Buy Tez" should put after "Fund from your wallet."

File: pages/[walletAddress]/fund-wallet.tsx [hunk 5]
```diff
@@ -38,15 +94,38 @@ const TopUpPage = () => {
...
+                create a transaction from your wallet to your TzSafe wallet
+              </p>
+              <button
+                className="mt-4 rounded bg-primary px-4 py-2 font-medium text-white hover:bg-red-500 hover:outline-none focus:bg-red-500"
+                onClick={() => {
+                  wertWidgetRef.current.mount();
+                  setError(undefined);
+                }}
+              >
+                Open the on-ramp service
```
[the remark this answers is not in the document]
Comment (rueshyna) [reply pages/[walletAddress]/fund-wallet.tsx:112]: - this button is on the left.
- the fund button is on the center.

**Review by rueshyna — CHANGES_REQUESTED** [review]
the warning message for not owner on this page should put under the title.

File: context/wert.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+}: {
+  wallet: string;
+  onSuccess: (txId: string) => void;
+}) =>
+  new WertWidget({
+    address: wallet,
+    partner_id: "01HBARVR2HGGY24WC52R4J89R8",
+    origin: "https://sandbox.wert.io",
+    network: "ghostnet",
+    theme: "dark",
```
Comment (rueshyna) [inline context/wert.ts:15]: the "default" one looks not bad. at least "X" is clear. :thinking:

File: context/wert.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+export const makeWertWidget = ({
+  wallet,
+  onSuccess,
+}: {
+  wallet: string;
+  onSuccess: (txId: string) => void;
+}) =>
+  new WertWidget({
+    address: wallet,
+    partner_id: "01HBARVR2HGGY24WC52R4J89R8",
```
[the remark this answers is not in the document]
Comment (JulesGuesnon) [reply context/wert.ts:12]: Yes we can, if anyone else steal it, they'll be able to use wert, but all the fees of the transactions will be for Marigold

File: context/wert.ts (hunk 6 above)
Comment (JulesGuesnon) [reply context/wert.ts:15]: Yeah but our theme is dark mode, so it wouldn't make sense to have a light mode I think

File: pages/[walletAddress]/fund-wallet.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (JulesGuesnon) [reply pages/[walletAddress]/fund-wallet.tsx:99]: - the `h2` have the same style, so I don't know which one you're referring to
- I not sure to understand what you mean by `after Fund from your wallet` as this text doesn't exist

File: pages/[walletAddress]/fund-wallet.tsx (hunk 2 above)
Comment (JulesGuesnon) [reply pages/[walletAddress]/fund-wallet.tsx:123]: I'm not sure to understand, do you mean showing the TzSafe alias?

File: context/wert.ts (hunk 6 above)
Comment (rueshyna) [reply context/wert.ts:15]: I just realized that our beacon-sdk doesn't use the same theme.
However, it's not the block to ship PR.
Close the thread.

**Review by rueshyna — APPROVED** [review]

**Pull request merged**